tasks/Task.cpp

```diff
@@ -13,37 +18,666 @@ Task::~Task()
 {
 }
 
-
-
 /// The following lines are template definitions for the various state machine
 // hooks defined by Orocos::RTT. See Task.hpp for more detailed
 // documentation about them.
 
 bool Task::configureHook()
 {
-    if (! TaskBase::configureHook())
+    try {
+        if (!TaskBase::configureHook()) {
+            return false;
+        }
+
+        if (!connectToCamera()) {
+            return false;
+        }
+
+        if (!switchOverAccess()) {
+            return false;
+        }
+
+        if (!configureCamera()) {
+            return false;
+        }
+        return true;
+    }
+    catch (GenICam::GenericException& ge) {
+        LOG_ERROR_S << "GenICam exception thrown: " << ge.what() << endl;
         return false;
-    return true;
+    }
+    catch (std::exception& ex) {
+        LOG_ERROR_S << "Standard exception thrown: " << ex.what() << endl;
+        return false;
+    }
 }
+
 bool Task::startHook()
 {
-    if (! TaskBase::startHook())
+    if (!TaskBase::startHook()) {
         return false;
+    }
+    startCamera();
     return true;
 }
 void Task::updateHook()
 {
     TaskBase::updateHook();
+    acquireFrame();
 }
 void Task::errorHook()
 {
     TaskBase::errorHook();
+    m_device->StopStream();
 }
 void Task::stopHook()
 {
     TaskBase::stopHook();
+    m_device->StopStream();
 }
 void Task::cleanupHook()
 {
     TaskBase::cleanupHook();
+    m_system->DestroyDevice(m_device);
+    Arena::CloseSystem(m_system);
+}
+
+bool Task::connectToCamera()
+{
+    LOG_INFO_S << "Looking for camera" << endl;
+    m_ip = _ip.get();
+    try {
+        m_system = Arena::OpenSystem();
+    }
+    catch (GenICam::GenericException& ge) {
+        LOG_WARN_S << "Arena System was already on, closing it and reopening." << endl;
+        Arena::CloseSystem(m_system);
```

Comment:
Since you call connectToCamera from configureHook, The rock component's state machine guarantees that m_system is not initialized here (i.e. either was the initial value - which should be nullptr - or is reset to nullptr in the cleanupHook.

A.k.a. this catch is most probably going to crash the component.

Reply:
I will test it again, with the m_system = nullptr because when I tried to restart the component there was an error because the Area::OpenSystem() was already called. I'll test removing this try and check if ti works now.